homeassistant/components/motioneye/__init__.py

```diff
@@ -429,6 +466,69 @@ async def handle_webhook(
     return None
 
 
+def _get_media_event_data(
+    hass: HomeAssistant,
+    device: dr.DeviceEntry,
+    event_file_path: str,
+    event_file_type: int,
+) -> dict[str, str]:
+    config_entry_id = next(iter(device.config_entries), None)
+    client = hass.data[DOMAIN].get(config_entry_id, {}).get(CONF_CLIENT)
+    coordinator = hass.data[DOMAIN].get(config_entry_id, {}).get(CONF_COORDINATOR)
+
+    if not coordinator or not client:
+        return {}
+
+    for identifier in device.identifiers:
+        data = split_motioneye_device_identifier(identifier)
+        if data is not None:
+            camera_id = data[2]
+            camera = get_camera_from_cameras(camera_id, coordinator.data)
+            break
+    else:
+        return {}
+
+    root_directory = camera.get(KEY_ROOT_DIRECTORY) if camera else None
+    if root_directory is None:
+        return {}
+
+    kind = "images" if client.is_file_type_image(event_file_type) else "movies"
+
+    # The file_path in the event is the full local filesystem path to the
+    # media. To convert that to the media path that motionEye will
+    # understand, we need to strip the root directory from the path.
+    if os.path.commonprefix([root_directory, event_file_path]) == root_directory:
+        file_path = "/" + os.path.relpath(event_file_path, root_directory)
+        output = {
+            EVENT_MEDIA_CONTENT_ID: (
+                f"{URI_SCHEME}{DOMAIN}/{config_entry_id}#{device.id}#{kind}#{file_path}"
+            ),
+        }
+        url = get_media_url(
+            client,
+            camera_id,
+            file_path,
+            kind == "images",
+        )
+        if url:
+            output[EVENT_FILE_URL] = url
+        return output
+
+    return {}
+
+
+def get_media_url(
+    client: MotionEyeClient, camera_id: int, path: str, image: bool
+) -> str | None:
+    """Get the URL for a motionEye media item."""
+    try:
+        if image:
+            return client.get_image_url(camera_id, path)
+        return client.get_movie_url(camera_id, path)
+    except MotionEyeClientPathError:
+        return None
```

Comment:
```suggestion
    with contextlib.suppress(MotionEyeClientPathError):
        if image:
            return client.get_image_url(camera_id, path)
        return client.get_movie_url(camera_id, path)
    return None
```

Doesn't really save much, but we usually write these like this